flang/include/flang/Parser/parse-tree.h

```diff
@@ -3738,6 +3738,29 @@ struct OmpOrderClause {
   std::tuple<std::optional<OmpOrderModifier>, Type> t;
 };
 
+// InteropType
+struct InteropType {
+  ENUM_CLASS(Kind, Target, TargetSync)
+  WRAPPER_CLASS_BOILERPLATE(InteropType, Kind);
+};
+
+// struct InteropPreference {
+//   UNION_CLASS_BOILERPLATE(InteropPreference);
+//   std::variant<Name,IntConstantExpr> u;
+// };
```

Comment:
For string literals, I think we can't use name here, Name is only used for storing Identifiers like variable name.

Use something like CharExpr or similar.